engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -595,9 +596,32 @@ namespace dmGameObject
             luaL_checktype(L, 3, LUA_TTABLE);
             lua_pushvalue(L, 3);
 
+            lua_getfield(L, -1, "key");
+            if (!lua_isnil(L, -1))
+            {
+                if (lua_isstring(L, -1))
+                {
+                    property_options.m_Key = dmHashString64(lua_tostring(L, -1));
+                }
+                else if (dmScript::IsHash(L, -1))
+                {
+                    property_options.m_Key = dmScript::CheckHash(L, -1);
+                }
```

Comment:
I think it's enough with "key = dmScript::CheckHashOrString(L, -1)" as it also handles error reporting

Reply:
soooo much better! thx